lib/p2p/NodeList.ts

```diff
@@ -87,6 +103,28 @@ class NodeList {
     return false;
   }
 
+  /**
+   * Add a reputation event to the nodes history
+   * @return true if the node was banned, false otherwise
+   */
+  public addReputationEvent = async (nodePubKey: string, type: ReputationEventType) => {
+    const node = this.nodes.get(nodePubKey);
+    if (node) {
+      node.reputationScore += ReputationEventWeight[type];
+
+      if (this.nodeIsBanned(node)) {
+        this.emit('node.ban', nodePubKey, node.reputationEvents);
```

Comment:
I figure we should only be sending events with negative scores for the ban reasons.

Reply:
After thinking a bit about this: why should we bother telling a node why we banned them? The banned node isn't important anymore and should just be informed about the fact that it got banned.

Reply:
@moshababo can probably answer this better than I can, but I think it's useful for debugging/testing purposes and also so that a node that was banned for honest reasons (something like bad connectivity) they can know what the problem is and possibly correct it. How would a peer know why it got banned without us telling it?